### ISSUE TITLE
Abstraer Notebooks de extracción de características a clases/Transformers

### ISSUE THREAD

**Issue by AdrianArnaiz** [AdrianArnaiz/TFG-Neurodegenerative-Disease-Detection#24]
Migrar el código de extracción de caracterísitcas utilizando disvoice a transformers siguiendo los estándares que sigue scikitlearn.

**Comment by AdrianArnaiz** [comment]
Entender correctamente el funcionamiento de los transformers viendo ejemplos como las prácticas de minería, documentacion scikitlearn...

**Comment by AdrianArnaiz** [comment]
## Funciones carga de Datos:
**Creación de funciones de carga de datos siguiendo los estándares que se siguen en sklearn/datasets/base.py**, es decir, siguiendo la manera en que lo hace el paquete sklearn.datasets.

Hay un archivo *cargaDatos.py* que contiene las funciones que cargan cada subset de características y devuelve un objeto de tipo Bunch con las variables data y target de la misma manera que se hace en sklearn.datasets (i.e load_iris()).

Las funciones se llamarán load_nombreSubsetCcas. Por ejemplo:
* `load_fon_rt(), load_fon_w_petaka(), , load_fon_v_A(), , load_art_w_petaka()`

**Comment by AdrianArnaiz** [comment]
## Clase ExtractorCaracterísticas 
Hemos creado la clase **ExtractorCaracterísticas** en el módulo extractorCcas que contiene **métodos** útilies para la extracción de características con Disvoice. Incluye métodos para añadir atributos extra. Estos atributos extra pueden ser añadidos a la vez que se extraen las demás características o pueden ser añadidos posteriormente a las correspondientes matrices numpy (utilizando add_atribs). Útil esto último si ya tenemos las ccas de Disvoice sacadas para no tener que volver a sacarlas. En resumen puede realizar las siguientes tareas: 
1. Extracción ccas Disvoice sin atributos Extra   
2. Extracción ccas Disvoice con atributos Extra   
3. Adicción atributos extra a ccas ya extraídas   
        
**Métodos:**   
* **Constructor**: se pasa como argumentos:   
  * la ruta donde se encuentran los audios (donde se encuentran las subcarpetas pd y hc)   
  * la ruta donde guardar los archivos de características generados por Disvoice    
  * opcionalmente el diccionario donde se tienen atributos extra de cada audio (edad,sexo...)   
* **extraccion_ccas_directorio**: Extrae las características de los audios en la ruta pasada en el constructor. tiene como parámetros:   
  * Script Disvoice usado para extraer las ccas [articulation, phonatio, prosody].   
  * Nombre del archivo donde guardar las ccas de los pacientes HC.   
  * Nombre del archivo donde guardar las ccas de los pacientes PD.   
  * Atributos_Extra (list)[opcional]: atributos a añadir a las ccas que deben estar incluidos en el diccionario pasado en el constructor. Si no se pasa nada, no se incluye ningun atributo extra.   
*  **add_atribs**: Añade los atributos a la matriz de ccas pasada, cuyos nombres se pasan en formato string y cuyo valor se encuentra en el diccionario pasado en el constructor. Éste método es útil para añadir los nuevos atributos a ccas ya extraídas. En ese caso, no es necesario que la ruta de los audios pasada en el constructor coincida con la matriz numpy que le pasamos, ya que no es usada. Únicamente añadimos las columnas necesarias a la matriz pasada.   
   * ccas: matriz a la que añadir los atributos.   
   * atribs: lista que contiene los atributos a añadir. Contiene el nombre de los atributos que se debe corresponder con el nombre que tienen en el diccionario pasado en el constructor.

**Comment by AdrianArnaiz** [comment]
## Más funciones carga de Datos:
Creación de funciones de carga de datos **PARA LOS NUEVOS CONJUNTOS DE CARACTERISTICAS #31 Y #32.** Al igual que antes, se han realizado siguiendo los estándares que se siguen en sklearn/datasets/base.py, es decir, siguiendo la manera en que lo hace el paquete sklearn.datasets. La carga para los diferentes tipos de datos está dividida en módulos python:
* **cargaDatos.py**: Datos directos Disvoice #19 
* **cargaDatosEdSx.py**:  Datos de Disvoice+ edad y sexo #31 
* **cargaDatosHombres.py**: Datos de Disv + Edad para audios de hombres. #32
* **cargaDatosMujeres.py**: Datos de Disv + Edad para audios de mujeres. #32
* **cargaDatosOrzOn.py**: Datos exactos articulo Orozco 2016 (Transiciones onset) #32
* **cargaDatos.py**: Datos exactos articulo Orozco 2016 (Transiciones offset) #32

**Comment by AdrianArnaiz** [comment]
## Añadido otro cambio a función gridSearchPipe #24 
Se ha añadido la normalización al pipeline en vez de hacerla en todo el conjunto de datos X.

Añadimos al pipe o le creamos. Explicado en notebooks y clase:

     if normalizar:
        if isinstance(pipe, Pipeline):
            pipe.steps.insert(0,['norm',MinMaxScaler()])
        else:
            pipe = Pipeline([('norm', MinMaxScaler()), ('clf', pipe)])
            if isinstance(pg, list):
                #para pasarle a dict: algunos paramgrids les definimos como [dict(params)]
                pg=pg[0]    
            for k in pg:
                pg['clf__'+k]=pg.pop(k)`
